sdk/keyvault/azure-security-keyvault-keys/src/key_client.cpp

```diff
@@ -30,82 +30,29 @@ struct RequestWithContinuationToken final
 };
 
 static inline RequestWithContinuationToken BuildRequestFromContinuationToken(
-    Azure::Security::KeyVault::Keys::GetPropertiesOfKeysOptions const& options,
-    std::vector<std::string>&& defaultPath)
+    const Azure::Nullable<std::string>& NextPageToken,
+    const Azure::Nullable<int32_t>& MaxPageResults,
+    const std::vector<std::string>&& defaultPath)
```

Comment:
```suggestion
    const std::vector<std::string>& defaultPath)
```

Reply:
@Jinming-Hu please help me understand the subtle nuances/difference here with your suggestion.

Reply:
const rvalue reference is very unusual and rarely used. I don't know when const rvalue ref can be useful.

@CaseyCarter Can you help us out here?

Reply:
I'll leave it as is for now since that's the current implementation in main, and will follow-up later.

Reply:
The reason is simple, to allow creation of unique_ptr from const pointer. Remember that const rvalue reference is more generic than just rvalue reference as it accepts both const and non-const. So we can allow the following:

int* const ptr = new int(9);
auto p = std::unique_ptr<int> { std::move(ptr) };

https://stackoverflow.com/questions/4938875/do-rvalue-references-to-const-have-any-use/60587511#60587511

the rvalue allow us to call the method passing a temp value, either const or non-const.  At any case, the method won't be able to mutate the value and the value will only live inside the method :)

Reply:
@vhvb1989 I'm not sure I can agree with you.


Just for this single case, how `const std::vector<std::string>&&` is better than `const std::vector<std::string>&`?
const rvalue ref can only bind to rvalue, but const lvalue ref can bind to both rvalue and lvalue.

Moreover, since we're going to make a copy of this parameter at Line 38 anyway, I think a by-value parameter is better than const reference.

Reply:
> Moreover, since we're going to make a copy of this parameter at Line 38 anyway, I think a by-value parameter is better than const reference.

A by-value parameter along with changing line 38 to a move assignment, yes.

Reply:
The proposed suggestion makes sense, but unrelated to this PRs goal, so I will leave it as a follow-up issue.